### PR TITLE
Rename API BranchCheckout to Checkout + add API support to checkout f…

### PR DESCRIPTION
**Pull request by freedomp**
…ile/s (+Native Git implementation)

Add Rebase command API
add Repository state to Status API (as String)
add support to set/get revision range (since, until) in Log API
-- Added above API changes to native git implementation, with empty implementation

Signed-off-by: Dror Cohen <dror.cohen@sap.com>

### CODE REVIEW

**Comment by skabashnyuk** [comment]
Eclipse CLA confirmed for dror.cohen@sap.com Eclipse CLA confirmed

File: platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitService.java [hunk 1]
```diff
@@ -111,12 +113,12 @@ public void add(AddRequest request) throws ApiException {
         }
     }
 
-    @Path("branch-checkout")
+    @Path("checkout")
```
Comment (mkuznyetsov) [inline platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitService.java]: It needs to rename this path in GitServiceClientImpl too

**Comment by skabashnyuk** [comment]
Could you please also rebase current master in your PR

File: platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitService.java (hunk 1 above)
Comment (freedomp) [inline platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitService.java]: Added a commit which fixes the path in GitServiceClientImpl

**Pull request merged**